EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -425,3 +433,57 @@ int RawDataUnpacker::ProcessVFATDataParallel(const uint16_t *buf, unsigned int m
 
   return wordsProcessed;
 }
+
+//----------------------------------------------------------------------------------------------------
+
+int RawDataUnpacker::ProcessOptoRxFrameSampic(const word *buf, unsigned int frameSize, TotemFEDInfo &fedInfo, SimpleVFATFrameCollection *fc) const
+{
+  
+  
+  unsigned int OptoRxId = (buf[0] >> 8) & 0xFFF;
+//   std::cout<< "Processing sampic frame: OptoRx " << OptoRxId << "  GOH " << gohIdx << "  Idx " << fiberIdx << "   framesize: " << frameSize << std::endl;
+
+  unsigned int orbitCounterVFATFrameWords = 6;
+  unsigned int sizeofVFATPayload = 12;
+  
+  VFATFrame::word *VFATFrameWordPtr = (VFATFrame::word *) buf;
```

Comment:
"const qualifier was removed via a cast, this may result in thread-unsafe code", see https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22796/27190/llvm-analysis/report-56957e.html#EndPath